lib/trollop.rb

```diff
@@ -169,16 +169,22 @@ def opt name, desc="", opts={}
       when Date; :date
       when Array
         if opts[:default].empty?
-          raise ArgumentError, "multiple argument type cannot be deduced from an empty array for '#{opts[:default][0].class.name}'"
-        end
-        case opts[:default][0]    # the first element determines the types
-        when Integer; :ints
-        when Numeric; :floats
-        when String; :strings
-        when IO; :ios
-        when Date; :dates
+          if opts[:type]
+            raise ArgumentError, "multiple argument type must be plural" unless MULTI_ARG_TYPES.include?(opts[:type])
+            nil
```

Comment:
Should this be returning opts[:type] instead of nil, so that the verification check afterwards can run?  I guess it doesn't really make a difference, because you are already doing the check here, but it makes the value of type_from_default more accurate.